simulator/gdb/gdb_interface.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * gdb_interface.c - Functional simulator interfaces for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+
+/* BFD config */
+#include <config.h>
+
+/* Simulator Interface */
+#include "sim_wrapper.h"
+
+/* GDB Interface */
+#include <gdb/remote-sim.h>
+#include <gdb/callback.h>
+/* GDB Simulator utilities */
+#include <sim-config.h>
+#include <sim-types.h>
+#include <sim-inline.h>
+#include <sim-arange.h>
+#include <sim-base.h>
+
+
+SIM_DESC sim_open (SIM_OPEN_KIND kind, struct host_callback_struct *callback,
+                   struct bfd *abfd, char *const *argv) {
+    SIM_DESC sd = sim_state_alloc (kind, callback);
+    if (!abfd)
+        fprintf (stderr, "Input file not set; please re-run GDB with input file");
```

Comment:
I do not see difference for return between sucessful and unsuccessful result.

Reply:
This will be fixed.